src/atom.rs

```diff
@@ -541,22 +536,140 @@ where
 
 ///////////////////////////////////////////////////////////////////////////////
 
-fn encode_date_time(dt: &DateTime<Utc>) -> BytesText<'static> {
-    let s = dt.to_rfc3339_opts(chrono::SecondsFormat::Millis, true);
+fn encode_timestamp(
+    col: &Arc<dyn Array>,
+    index: usize,
+    unit: TimeUnit,
+    tz: Option<Arc<str>>,
+) -> Result<BytesText<'static>, UnsupportedDataType> {
+    let dt = match unit {
+        TimeUnit::Microsecond => {
+            let value = cast_primitive::<TimestampMicrosecondType>(col, index)?;
+            DateTime::from_timestamp_micros(value)
+        }
+        TimeUnit::Millisecond => {
+            let value = cast_primitive::<TimestampMillisecondType>(col, index)?;
+            DateTime::from_timestamp_millis(value)
+        }
+        TimeUnit::Nanosecond => {
+            let value = cast_primitive::<TimestampNanosecondType>(col, index)?;
+            Some(DateTime::from_timestamp_nanos(value))
+        }
+        TimeUnit::Second => {
+            let value = cast_primitive::<TimestampSecondType>(col, index)?;
+            DateTime::from_timestamp(value, 0)
+        }
+    };
+
+    match dt {
+        Some(d) => Ok(encode_date_time(&d, tz.is_some())),
+        None => Err(UnsupportedDataType::new(DataType::Timestamp(unit, tz))),
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+fn encode_date_time(dt: &DateTime<Utc>, tz: bool) -> BytesText<'static> {
+    let s = if tz {
+        dt.to_rfc3339_opts(chrono::SecondsFormat::Millis, true)
+    } else {
+        dt.naive_utc().format("%Y-%m-%dT%H:%M").to_string()
```

Comment:
Shouldn't this be `%Y-%m-%dT%H:%M:%S%.f` not to lose seconds and fractional precision?